plugins/resources/src/unixfilesystem.cpp

```diff
@@ -83,6 +87,198 @@
 const std::string DEFAULT_VAULT_DIR_MODE( "default_vault_directory_mode_kw" );
 const std::string HIGH_WATER_MARK( "high_water_mark" ); // no longer used
 const std::string REQUIRED_FREE_INODES_FOR_CREATE("required_free_inodes_for_create"); // no longer used
+const std::string HOST_MODE("HOST_MODE");
+const std::string HOST_LIST("HOST_LIST");
+
+namespace {
+    bool is_operating_in_detached_mode(irods::plugin_property_map& prop_map)
+    {
+
+        std::string host_mode_str;
+
+        irods::error ret = prop_map.get< std::string >(HOST_MODE, host_mode_str);
+        if (ret.ok()) {
+            if (boost::iequals(host_mode_str.c_str(), "detached"))  {
+                return true;
+            }
+        }
+
+        // default is attached mode
+        return false;
+    }
+}
+
+static std::vector<std::string> split(const std::string str, char delim)
+{
+    std::vector<std::string> result;
+    std::istringstream ss{str};
+    std::string token;
+    while (std::getline(ss, token, delim)) {
+        if (!token.empty()) {
+            result.push_back(token);
+        }
+    }
+    return result;
+}
+
+// returns a tuple with the following:
+//   bool - true iff the resource_hostname is in HOST_LIST string
+//   std::optional<std::string> - the new vault path if it is defined
+std::tuple<bool, std::optional<std::string>> get_detached_mode_vault_path(
```

Comment:
You might want to add a comment here about the forward declaration in the unit test, as if the function signature is ever changed here, it will also need to be changed in the unit test.

Reply:
done